src/dtx/dtx_common.c

```diff
@@ -525,15 +525,16 @@ int
 dtx_leader_end(struct dtx_leader_handle *dlh, struct ds_cont_child *cont,
 	       int result)
 {
-	struct dtx_handle		*dth = &dlh->dlh_handle;
+	struct dtx_handle		*dth;
 	struct dtx_conflict_entry	*dces = NULL;
 	int				*ptr = NULL;
 	int				 dces_cnt = 0;
 	int				 flags = 0;
 	int				 rc = 0;
 
-	if (dlh == NULL)
-		return result;
+	D_ASSERT(dlh != NULL);
+	dth = &dlh->dlh_handle;
+	D_ASSERT(dth != NULL);
```

Comment:
struct dtx_leader_handle {
        /* The dtx handle on the leader node */
        struct dtx_handle               dlh_handle;
        /* result for the distribute transaction */
        int                             dlh_result;
...
}

The "dtx_leader_handle::dlh_handle" will never be NULL, so the check is redundant.

Reply:
Indeed, so I think I was wrong, it's not the root cause of the segfault.